File: 1-easy/00027-remove-element/two-pointers.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {

        size_t len = nums.size();
        size_t k = 0;

        for (size_t i{0}; i < len; ++i) {
            if (nums[i] != val) {
                nums[k] = nums[i];
                k += 1;
            }
        }

        return k;
    }
};

File: 1_easy/00027_remove_element/two_pointers.rs
impl Solution {
    pub fn remove_element(nums: &mut Vec<i32>, val: i32) -> i32 {

        let mut len = nums.len();
        let mut k = 0_usize;

        for i in 0..len {
            if nums[i] != val {
                nums[k] = nums[i];
                k += 1;
            }
        }

        k as i32
    }
}
